List partial causatives

dnil opened the pull request:
This PR adds a functionality or fixes a bug.

When marking partial causative, variant can be seen on causatives view
![Screenshot 2022-03-08 at 10 29 35](https://user-images.githubusercontent.com/758570/157209057-5d72828a-16de-4f92-8eae-6b74c0d67c69.png)
and as a matching causative
![Screenshot 2022-03-08 at 10 38 44](https://user-images.githubusercontent.com/758570/157209935-cdc83675-c5e9-4200-ab27-31d7d99d556b.png)
also, on the variant level:
![Screenshot 2022-03-08 at 11 23 27](https://user-images.githubusercontent.com/758570/157217648-64c28b26-84dd-4c78-aa09-9ec13f5d420a.png)
 
compared to main:
![Screenshot 2022-03-08 at 10 41 25](https://user-images.githubusercontent.com/758570/157210282-048c80ea-a9e6-472c-ac56-825158e93c34.png)
![Screenshot 2022-03-08 at 10 41 19](https://user-images.githubusercontent.com/758570/157210286-e7290fb4-e176-47e4-ae36-7985de601c3e.png)
Especially the latter is a little worrying, but afaik at least cust002 rarely use partial.

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [ ] tests executed by

northwestwitch reviewed (approved):
Super useful fix! 🏅 

You might want to check some vulture complaints in mongo/variant.py.

I would make partial causative badge a bit more evident by changing its color and moving it at the beginning of the row. Add also a tooltip to explain what is a partial causative for those who are not already familiar with the concept?

northwestwitch commented:
What happens in the eventuality that a variant is a causative in a case and a partial causative in another? 🤔

dnil commented on scout/adapter/mongo/variant.py:
```diff
@@ -578,7 +577,11 @@ def check_causatives(self, case_obj=None, institute_obj=None, limit_genes=None):
-            {"institute": institute_id, "verb": "mark_causative", "category": "variant"}
+            {
+                "institute": institute_id,
+                "verb": {"$in": ["mark_causative", "mark_partial_causative"]},
```
Something like that; we actually discussed when introducing the feature and postponed it until it saw more use (https://github.com/Clinical-Genomics/scout/pull/1390).

dnil commented on scout/server/blueprints/institutes/templates/overview/causatives.html:
```diff
@@ -142,6 +142,9 @@
             </td>
             <td>
+              {% if (case.partial_causatives and variant._id in case.partial_causatives) %}
+                <span class="badge badge-secondary">partial</span>
```
Hm, both the case status and the partial in that case - they belong together. I would say they are good where they are, but explain if Im missing a use case!

I'm no colouring expert so will not press it but my two cents are: 1) its good to keep the number of colors down, much like the number of fonts, but 2) colors should not visually collide with any that is used for quick recognition in the page
3) less important features should have lower signal/contrast value 4) the color should stay roughly the same between pages. Its `badge-secondary` in e.g. the general report. It is not much of a difference when considering a large list, it does not need a high contrast value. 

But, honestly, any of your suggestions is good enough though; I could change them all to e.g. `badge-warning`-yellow instead? Or just keep them for now. 🤷

dnil replied:
I guess I should make a css class instead really. `badge-partial-causative`. 😁

dnil commented:
> What happens in the eventuality that a variant is a causative in a case and a partial causative in another? 🤔

That should be fine; both (variant, case)-tuples should show in the list. I'll check to make sure!

dnil commented:
Thanks for the review! I changed the color, added tooltips and fixed Vulture suggestions. 

![Screenshot 2022-03-08 at 14 20 10](https://user-images.githubusercontent.com/758570/157246231-507c7b5f-2d09-45a3-b513-7c218ea3b5c5.png)
![Screenshot 2022-03-08 at 14 20 38](https://user-images.githubusercontent.com/758570/157246234-e0a8920b-6819-4c06-b91b-f13b470c41c7.png)
